server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -266,7 +266,7 @@ public void afterIndexRemoved(Index index, IndexSettings indexSettings, IndexRem
         // it's fine to keep the contexts open if the index is still "alive"
         // unfortunately we don't have a clear way to signal today why an index is closed.
         // to release memory and let references to the filesystem go etc.
-        if (reason == IndexRemovalReason.DELETED || reason == IndexRemovalReason.CLOSED) {
+        if (reason == IndexRemovalReason.DELETED || reason == IndexRemovalReason.CLOSED || reason == IndexRemovalReason.REOPENED) {
```

Comment:
might not be necessary, because we have no search contexts when an index is closed. But better safe than sorry.

Reply:
That's also what I think.